WrapImage/nifti_wrapper.py

```diff
@@ -75,52 +24,162 @@ def loop_over_first_n_minus_1_dimensions(arr):
         flat_view = arr[idx].flatten()
         yield idx, flat_view
 
+def generate_data(data, bvals, b0_indices, groups, total_iteration):
```

Comment:
1: Is b0_indices something you want to tell the algorithm, or is it more fool proof if we generate them locally from bvals?
2: Maybe we need to start documenting some code; for example, to me "generate_data" sounds like it will generate data, but it is unclear why it needs data for this (if there is already data, why generate it?)

Reply:
The source could come from anywhere, this splits the data up based on the bvals and directions. The b0 indices are included in order to give the b0 data to every fit.

True, I should update it with some more documentation.